$OpenBSD: patch-daemon_gdm-display-access-file_c,v 1.2 2020/11/14 11:45:33 ajacoutot Exp $

The X(7) server runs as the _x11 user, so give it read access to the
authorization file to prevent:
(EE) Failed to open authorization file "/var/run/gdm/auth-for-_gdm-XXXXXX/database": Permission denied

Index: daemon/gdm-display-access-file.c
--- daemon/gdm-display-access-file.c.orig
+++ daemon/gdm-display-access-file.c
@@ -337,9 +337,15 @@ _create_xauth_file_for_user (const char  *username,
         g_debug ("GdmDisplayAccessFile: creating %s", auth_filename);
         /* mode 00600 */
         errno = 0;
-        fd = g_open (auth_filename,
-                     O_RDWR | O_CREAT | O_EXCL | O_BINARY,
-                     S_IRUSR | S_IWUSR);
+        if (uid == 676) {
+                fd = g_open (auth_filename,
+                             O_RDWR | O_CREAT | O_EXCL | O_BINARY,
+                             S_IRUSR | S_IWUSR | S_IRGRP);
+        } else {
+                fd = g_open (auth_filename,
+                             O_RDWR | O_CREAT | O_EXCL | O_BINARY,
+                             S_IRUSR | S_IWUSR);
+        }
 
         if (fd < 0) {
                 g_set_error (error,
